client_linux.go

```diff
@@ -55,23 +57,45 @@ func (c *client) Devices() ([]*Device, error) {
 
 // Ports implements osClient.
 func (c *client) Ports() ([]*Port, error) {
-	msgs, err := c.execute(unix.DEVLINK_CMD_PORT_GET, netlink.Dump)
+	msgs, err := c.execute(unix.DEVLINK_CMD_PORT_GET, netlink.Dump, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	return parsePorts(msgs)
 }
 
+// DpipeTables implements osClient.
+func (c *client) DpipeTables(dev *Device) ([]*DpipeTable, error) {
+	if dev == nil {
+		return nil, fmt.Errorf("invalid argument")
+	}
+	encoder := netlink.NewAttributeEncoder()
```

Comment:
Nit: I usually call this `ae` per convention.

Reply:
Fixed.